### ISSUE TITLE
Jobs with errors are not removed from the list of current jobs

### ISSUE THREAD

**Issue by ntBre** [ntBre/psqs#5]
The error branch prints a warning and increments the number of failed jobs, meaning that these jobs will continue being polled, printed, and counted on each iteration of the loop. Some users would prefer these jobs to be resubmitted (in which case the if and else collapse into the same branch), but for the current behavior, these jobs should simply be counted once and removed from the polling list.

https://github.com/ntBre/psqs/blob/d2ea8fe8cc2b4efacb20c33f6a7f5a0b70d30842/src/queue/drain.rs#L178

**Comment by ntBre** [comment]
On the other hand, continuing to poll these jobs is useful if the user manually inspects the output file and determines that it should be resubmitted. They can trigger the resubmission by deleting the output file. Still, the jobs should not be counted or warned about on every iteration. Maybe a `has_errored` field should be added to `Job` to prevent this.

**Comment by ntBre** [comment]
I think a `has_errored` field is a very good idea. Otherwise, the manually deleted output files don't reset the `failed_jobs` counter, causing the function to return an error even if everything worked out in the end.

**Comment by ntBre** [comment]
This was resolved starting with 9005691. `failed_jobs` was changed from a counter to a set to avoid double-counting the same jobs. I decided *not* to collapse the branches to allow the "trigger the resubmission by deleting the output file" behavior described above but to avoid resubmitting the same failing jobs on loop automatically. For now, I think this is the best balance between giving the user some ability to resubmit jobs they know have failed spuriously and not spamming the queuing system with syntax errors, for example. However, I just noticed that a warning is still printed on each iteration, so I will push one more fix before closing this issue.
